pkg/apis/gcp/v1alpha1/types_infrastructure.go

```diff
@@ -101,6 +101,8 @@ type VPC struct {
 	// CloudRouter indicates whether to use an existing CloudRouter or create a new one
 	// +optional
 	CloudRouter *CloudRouter `json:"cloudRouter,omitempty"`
+	// EnablePrivateGoogleAccess enables PrivateGoogleAccess for the workers subnet.
+	EnablePrivateGoogleAccess bool `json:"enablePrivateGoogleAccess,omitempty"`
```

Comment:
You could add end users usage doc to `./docs/usage-as-end-user.md` - explain with few words what the fields does and add it to the example InfrastructureConfig.